charts_plugin/src/main/java/com/futureworkshops/mobileworkflow/plugin/charts/view/ItemProvider.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022 FutureWorkshops. All rights reserved.
+ */
+
+package com.futureworkshops.mobileworkflow.plugin.charts.view
+
+import com.futureworkshops.mobileworkflow.data.network.task.URLIAsyncTask
+import com.futureworkshops.mobileworkflow.domain.service.ServiceContainer
+import com.futureworkshops.mobileworkflow.plugin.charts.step.DashboardBaseChartItem
+import com.futureworkshops.mobileworkflow.plugin.charts.step.DashboardChartItem
+import com.futureworkshops.mobileworkflow.plugin.charts.step.DashboardNetworkChartItem
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+sealed class ItemProvider {
+
+    abstract fun onItemsReady(
+        provider: (List<DashboardBaseChartItem>) -> Unit,
+        onLoading: () -> Unit,
+        onError: () -> Unit
+    )
+
+    open fun retry() {}
+
+    class SyncItemsProvider(private val items: List<DashboardChartItem>) : ItemProvider() {
+        override fun onItemsReady(
+            provider: (List<DashboardBaseChartItem>) -> Unit,
+            onLoading: () -> Unit,
+            onError: () -> Unit
+        ) {
+            val formalisedItems = items.map {
+                DashboardBaseChartItem(
+                    id = it.id,
+                    listItemId = it.listItemId,
+                    chartType = it.chartType,
+                    title = it.title,
```

Comment:
@jmcarcedo It is important to pass these strings through the localization services. Especially title and subtitle. To ensure that (even more for the static dashboard) these strings can be translated using the App Builder. The comment is also valid for the network response (from the load method) down below.

Reply:
Changed in [0179af8](https://github.com/FutureWorkshops/MWChartsPlugin-Android/pull/12/commits/0179af8b501acb4626143b73badd1e5a2c7765a9)